File: src/combined/Models/JsModels/donation.ts
export default class Donation {
  id: number;
  eventID: number;
  teamID?: number;
  personID?: number;
  date: string;
  amount: number;
  isPledge: boolean;

  constructor(
    eventID: number,
    date: string,
    amount: number,
    teamID?: number,
    personID?: number
  ) {
    this.eventID = eventID;
    this.teamID = teamID;
    this.personID = personID ?? undefined;
    this.date = date;
    this.amount = amount;
    this.isPledge = false;
    this.id = undefined;
  }
}

File: src/combined/Models/Entities/Donation.cs
﻿using System.ComponentModel.DataAnnotations.Schema;
using System.Transactions;

namespace Api.Models.Entities;

public record Donation
{
    public long ID { get; init; }
    public long EventID { get; init; }
    public Event Event { get; init; }
    public long? TeamID { get; init; }
    public Team Team { get; init; }
    public long? PersonID { get; init; }
    public Person Person { get; init; }
    public DateTime Date { get; init; }
    public decimal Amount { get; init; }
    public bool IsPending { get; init; } = true;
    public long? Link { get; init; }
    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid TransactionNumber  { get; init; }

    public string? AuthorizationNumber { get; init; }
}
